test/integration/version_integration_test.cc

```diff
@@ -1,52 +0,0 @@
-#include "test/integration/http_integration.h"
-
-namespace Envoy {
-namespace {
-
-// Integration test for ingestion of configuration across API versions.
-// Currently we only have static tests, but there will also be xDS tests added
-// later.
-class VersionIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
-                               public HttpIntegrationTest {
-public:
-  VersionIntegrationTest() : HttpIntegrationTest(Http::CodecType::HTTP1, GetParam()) {}
-};
-
-INSTANTIATE_TEST_SUITE_P(IpVersions, VersionIntegrationTest,
-                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()));
-
-// Just IP tagging for now.
-const char ExampleIpTaggingConfig[] = R"EOF(
-    request_type: both
-    ip_tags:
-      - ip_tag_name: external_request
-        ip_list:
-          - {address_prefix: 1.2.3.4, prefix_len: 32}
-)EOF";
-
-// envoy.filters.http.ip_tagging from v3 TypedStruct config.
-TEST_P(VersionIntegrationTest, IpTaggingV3StaticTypedStructConfig) {
-  config_helper_.prependFilter(absl::StrCat(R"EOF(
-name: ip_tagging
-typed_config:
-  "@type": type.googleapis.com/xds.type.v3.TypedStruct
-  type_url: type.googleapis.com/envoy.extensions.filters.http.ip_tagging.v3.IPTagging
-  value:
-  )EOF",
-                                            ExampleIpTaggingConfig));
-  initialize();
-}
-
-// envoy.filters.http.ip_tagging from v3 typed Any config.
-TEST_P(VersionIntegrationTest, IpTaggingV3StaticTypedConfig) {
-  config_helper_.prependFilter(absl::StrCat(R"EOF(
-  name: ip_tagging
-  typed_config:
-    "@type": type.googleapis.com/envoy.extensions.filters.http.ip_tagging.v3.IPTagging
-  )EOF",
-                                            ExampleIpTaggingConfig));
-  initialize();
-}
```

Comment:
Seems a bit different (uses the typed Any) then the test above (uses TypedStruct).
Why was this test removed? Was it because this file shouldn't test IpTagging, but related to generic filter configs?

Reply:
Because it wasn't the type integrationt est, it was the version integration test, making sure v2 and v3 worked.  I think we don't need that any more (but assigned it your way as the version czar in case  I'm wrong) and it resulted in non-core-dependendies in core locations

I'm happy to keep both tests over in iptagging if we think there's value in preserving both